app/src/main/java/com/teamagam/gimelgimel/app/model/entities/LocationSample.java

```diff
@@ -19,7 +21,7 @@
  * (UTC time), all other
  * parameters are optional.
  */
-public class LocationSample {
+public class LocationSample implements Parcelable {
```

Comment:
I know you spent some time on it but I would use Android Studio's basic Parcelable implementation.